packages/image_picker/image_picker_windows/lib/image_picker_windows.dart

```diff
@@ -0,0 +1,163 @@
+    'webp',
+    'gif',
+    'tif',
+    'tiff',
+    'apng'
+  ];
+
+  /// List of video extensions used when picking videos
+  @visibleForTesting
+  static const List<String> videoFormats = <String>[
+    'mov',
+    'wmv',
+    'mkv',
+    'mp4',
+    'webm',
+    'avi',
+    'mpeg',
+    'mpg'
+  ];
+
+  /// The FileSelectorPlatform used by the ImagePickerWindows.
+  @visibleForTesting
+  static late FileSelectorPlatform fileSelectorInstance = FileSelectorWindows();
+
+  /// Registers this class as the default instance of [ImagePickerPlatform].
+  static void registerWith() {
+    ImagePickerPlatform.instance = ImagePickerWindows();
+  }
+
+  // Note that the `maxWidth`, `maxHeight`, `imageQuality`
+  // and `preferredCameraDevice` arguments are not supported on Windows.
+  // If any of these arguments is supplied, it'll be silently ignored
+  // by the Windows version of the plugin. `source` is not implemented
+  // for `ImageSource.camera` and will throw an exception.
+  @override
+  Future<PickedFile?> pickImage({
+    required ImageSource source,
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+  }) async {
+    final XFile? file = await getImage(
+        source: source,
+        maxWidth: maxWidth,
+        maxHeight: maxHeight,
+        imageQuality: imageQuality,
+        preferredCameraDevice: preferredCameraDevice);
+    if (file != null) {
+      return PickedFile(file.path);
+    }
+    return null;
+  }
+
+  // Note that the `preferredCameraDevice` and `maxDuration`
+  // arguments are not supported on Windows. If any of these arguments is
+  // supplied, it'll be silently ignored by the Windows version of the
+  // plugin. `source` is not implemented for `ImageSource.camera` and
+  // will throw an exception.
+  @override
+  Future<PickedFile?> pickVideo({
+    required ImageSource source,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+    Duration? maxDuration,
+  }) async {
+    final XFile? file = await getVideo(
+        source: source,
+        preferredCameraDevice: preferredCameraDevice,
+        maxDuration: maxDuration);
+    if (file != null) {
+      return PickedFile(file.path);
+    }
+    return null;
+  }
+
+  // Note that the `maxWidth`, `maxHeight`, `imageQuality`, and
+  // `preferredCameraDevice` arguments are not supported on Windows. If
+  // any of these arguments is supplied, it'll be silently ignored by the
+  // Windows version of the plugin. `source` is not implemented for
+  // `ImageSource.camera` and will throw an exception.
+  @override
+  Future<XFile?> getImage({
+    required ImageSource source,
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+  }) async {
+    if (source != ImageSource.gallery) {
+      throw UnimplementedError(
+          'ImageSource.gallery is currently the only supported source on Windows');
```

Comment:
Is the intent to add other image sources in the future? If so, please file an issue to track this, add a `TODO` here and link the issue.

```
// TODO(username): Support ImageSource.foo, ImageSource.bar.
//                 See http://link/to/a/bug/123
```

If we will never support other image sources, throw `UnsupportedError` instead.